File: hawkey/tests/test_advisoryref.c
/*
 * Copyright (C) 2014 Red Hat, Inc.
 *
 * Licensed under the GNU Lesser General Public License Version 2.1
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 */


// hawkey
#include "src/advisory.h"
#include "src/advisoryref.h"
#include "src/package.h"
#include "fixtures.h"
#include "test_suites.h"
#include "testsys.h"

static HyAdvisoryRef reference;

static void
advisoryref_fixture(void)
{
    fixture_yum();

    HyPackage pkg;
    HyAdvisoryList advisories;
    HyAdvisory advisory;
    HyAdvisoryRefList reflist;

    pkg = by_name(test_globals.sack, "tour");
    advisories = hy_package_get_advisories(pkg, HY_GT);
    advisory = hy_advisorylist_get_clone(advisories, 0);
    reflist = hy_advisory_get_references(advisory);
    reference = hy_advisoryreflist_get_clone(reflist, 0);

    hy_advisoryreflist_free(reflist);
    hy_advisory_free(advisory);
    hy_advisorylist_free(advisories);
    hy_package_free(pkg);
}

static void
advisoryref_teardown(void)
{
    hy_advisoryref_free(reference);
    teardown();
}

START_TEST(test_type)
{
    ck_assert_int_eq(hy_advisoryref_get_type(reference), HY_REFERENCE_BUGZILLA);
}
END_TEST

START_TEST(test_id)
{
    ck_assert_str_eq(hy_advisoryref_get_id(reference), "472090");
}
END_TEST

START_TEST(test_title)
{
    ck_assert_str_eq(hy_advisoryref_get_title(reference), "/etc/init.d/clvmd points to /usr/sbin for LVM tools");
}
END_TEST

START_TEST(test_url)
{
    ck_assert_str_eq(hy_advisoryref_get_url(reference), "https://bugzilla.redhat.com/show_bug.cgi?id=472090");
}
END_TEST

Suite *
advisoryref_suite(void)
{
    Suite *s = suite_create("AdvisoryRef");
    TCase *tc;

    tc = tcase_create("WithRealRepo");
    tcase_add_unchecked_fixture(tc, advisoryref_fixture, advisoryref_teardown);
    tcase_add_test(tc, test_type);
    tcase_add_test(tc, test_id);
    tcase_add_test(tc, test_title);
    tcase_add_test(tc, test_url);
    suite_add_tcase(s, tc);

    return s;
}

File: dnf-4/python/hawkey/tests/tests/test_advisoryref.py
#
# Copyright (C) 2014 Red Hat, Inc.
#
# Licensed under the GNU Lesser General Public License Version 2.1
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2.1 of the License, or (at your option) any later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
#

"""Tests of the _hawkey.AdvisoryRef class."""

from __future__ import absolute_import
from __future__ import unicode_literals
from . import base

import hawkey
import itertools

def find_reference(sack, url):
    """Find an advisory reference with given URL."""
    # The function is needed because references cannot be retrieved directly.
    advisories_iterable = itertools.chain(
        (pkg.get_advisories(hawkey.LT) for pkg in hawkey.Query(sack)),
        (pkg.get_advisories(hawkey.GT | hawkey.EQ) for pkg in hawkey.Query(sack)))
    advisories = itertools.chain.from_iterable(advisories_iterable)
    references_iterable = (advisory.references for advisory in advisories)
    for reference in itertools.chain.from_iterable(references_iterable):
        if reference.url == url:
            return reference

class Test(base.TestCase):
    """Test case consisting of one random advisory reference."""

    def setUp(self):
        """Prepare the test fixture."""
        sack = base.TestSack(repo_dir=self.repo_dir)
        sack.load_repo(load_updateinfo=True)
        self.reference = find_reference(
            sack, 'https://bugzilla.redhat.com/show_bug.cgi?id=472090')

    def test_type(self):
        self.assertEqual(self.reference.type, hawkey.REFERENCE_BUGZILLA)

    def test_id(self):
        self.assertEqual(self.reference.id, '472090')

    def test_title(self):
        self.assertEqual(self.reference.title,
                         '/etc/init.d/clvmd points to /usr/sbin for LVM tools')

    def test_url(self):
        self.assertEqual(self.reference.url,
                         'https://bugzilla.redhat.com/show_bug.cgi?id=472090')
